drivers/sdhc/sdhci_common.c

```diff
@@ -699,7 +795,11 @@ static int sdhci_send_cmd(struct sdhci_common *sdhci_ctx, struct sdhc_command *c
 	sys_write16(SDHCI_MAKE_CMD(cmd->opcode, cmd_flags), sdhci_ctx->reg_base + SDHCI_COMMAND);
 
 	if (cmd->opcode != SD_SEND_TUNING_BLOCK) {
-		ret = sdhci_poll_cmd_complete(sdhci_ctx, cmd->timeout_ms);
+		if (sdhci_ctx->f_use_irq) {
+			ret = sdhc_wait_cmd_complete(sdhci_ctx, cmd->timeout_ms);
```

Comment:
The next comment isn't related to this PR (at least I think it isn't).

Should you introduce a retry mechanism to the driver in case of an error?
`sdhc_command` has a `retries` field for this purpose.

Reply:
i'll add it in brcm_bcm2712_sdhc